### PR TITLE
[rush-lib] Rename Task -> Operation

### PR DESCRIPTION
**Pull request by dmichon-msft**
## Summary
Renames `Task` to `Operation` to match the official documentation. An `Operation` is a unit of work to the Rush orchestrator, and the initial `OperationSelector` creates one for each selected `[RushConfigurationProject, IPhase]` tuple.

## Details
`Task` -> `Operation`
`ProjectTaskRunner` -> `ShellOperationRunner` (since it is an `IOperationRunner` implemented via shell command)
`ProjectTaskSelector` -> `OperationSelector`
etc.

Updates relevant variable names, code comments, etc.

## How it was tested
Local runs of `rush build -v -t rush-lib` with built code. All changes should be cosmetic, though test snapshots are affected due to the naming.

### CODE REVIEW

File: apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 1]
```diff
@@ -132,33 +131,33 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommand> {
...
+      phasesToRun: new Set(this._initialPhases)
     });
 
     const isWatch: boolean = this._watchParameter?.value || this._alwaysWatch;
 
     const executeOptions: IExecuteInternalOptions = {
-      taskSelector,
-      taskExecutionManagerOptions: taskExecutionManagerOptions,
+      operationSelector: operationSelector,
+      executionManagerOptions: executionManagerOptions,
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:158]:
```suggestion
      operationSelector,
      executionManagerOptions,
```

File: apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 2]
```diff
@@ -132,33 +131,33 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommand> {
...
       projectSelection,
-      taskFactoryOptions,
+      operationFactoryOptions: operationFactoryOptions,
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:160]:
```suggestion
      operationFactoryOptions,
```

File: apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 3]
```diff
@@ -333,22 +332,22 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommand> {
...
-    const taskFactory: ProjectTaskFactory = new ProjectTaskFactory(taskFactoryOptions);
+    const operationFactory: OperationFactory = new OperationFactory(operationFactoryOptions);
 
-    const taskExecutionManager: TaskExecutionManager = new TaskExecutionManager(
-      taskSelector.createTasks({
+    const executionManager: OperationExecutionManager = new OperationExecutionManager(
+      operationSelector.createOperations({
         projectSelection,
-        taskFactory
+        operationFactory: operationFactory
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:342]:
```suggestion
        operationFactory
```

File: apps/rush-lib/src/logic/operations/OperationSelector.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+}
+
+interface ISelectedOperationNode extends IOperationNode {
+  operation: Operation;
+}
+
+/**
+ * This class is responsible for transforming a set of selected phases and selected projects into an operation dependency graph.
+ */
+export class ProjectTaskSelector {
```
Comment (iclanton) [inline apps/rush-lib/src/logic/operations/OperationSelector.ts:44]:
```suggestion
export class OperationSelector {
```

File: common/changes/@microsoft/rush/rename-task-to-operation_2022-02-08-02-08.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "Rename \"Task\" to \"Operation\" in source code to conform to documentation.",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/rename-task-to-operation_2022-02-08-02-08.json:5]:
```suggestion
      "comment": "",
```

A user or API consumer isn't going to be affected by this change.

File: apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 6]
```diff
@@ -11,19 +11,19 @@ import { SetupChecks } from '../../logic/SetupChecks';
...
+} from '../../logic/operations/OperationExecutionManager';
 import { RushConstants } from '../../logic/RushConstants';
 import { EnvironmentVariableNames } from '../../api/EnvironmentConfiguration';
 import { LastLinkFlag, LastLinkFlagFactory } from '../../api/LastLinkFlag';
 import { RushConfigurationProject } from '../../api/RushConfigurationProject';
 import { BuildCacheConfiguration } from '../../api/BuildCacheConfiguration';
 import { SelectionParameterSet } from '../SelectionParameterSet';
 import type { CommandLineConfiguration, IPhase, IPhasedCommand } from '../../api/CommandLineConfiguration';
-import { ProjectTaskSelector } from '../../logic/ProjectTaskSelector';
+import { ProjectTaskSelector } from '../../logic/operations/OperationSelector';
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:24]:
```suggestion
import { OperationSelector } from '../../logic/operations/OperationSelector';
```

File: apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 7]
```diff
@@ -43,31 +43,30 @@ export interface IPhasedScriptActionOptions extends IBaseScriptActionOptions<IPh
 }
 
 interface IExecuteInternalOptions {
-  taskSelector: ProjectTaskSelector;
-  taskExecutionManagerOptions: ITaskExecutionManagerOptions;
+  operationSelector: ProjectTaskSelector;
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:46]:
```suggestion
  operationSelector: OperationSelector;
```

File: apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts [hunk 8]
```diff
@@ -132,33 +131,33 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommand> {
...
       quietMode: isQuietMode,
       debugMode: this.parser.isDebug,
       parallelism: parallelism,
       changedProjectsOnly: changedProjectsOnly,
       repoCommandLineConfiguration: this._repoCommandLineConfiguration
     };
 
-    const taskSelector: ProjectTaskSelector = new ProjectTaskSelector({
-      phasesToRun: new Set(this._initialPhase)
+    const operationSelector: ProjectTaskSelector = new ProjectTaskSelector({
```
Comment (iclanton) [inline apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts:150]:
```suggestion
    const operationSelector: OperationSelector = new OperationSelector({
```

File: apps/rush-lib/src/logic/test/ProjectTaskSelector.test.ts [hunk 9]
```diff
@@ -3,39 +3,39 @@
 
 import { RushConfiguration } from '../../api/RushConfiguration';
 import { CommandLineConfiguration, IPhasedCommand } from '../../api/CommandLineConfiguration';
-import { IProjectTaskOptions, IProjectTaskFactory, ProjectTaskSelector } from '../ProjectTaskSelector';
-import { Task } from '../taskExecution/Task';
+import { IOperationOptions, IOperationFactory, ProjectTaskSelector } from '../operations/OperationSelector';
```
Comment (iclanton) [inline apps/rush-lib/src/logic/test/ProjectTaskSelector.test.ts:6]:
```suggestion
import { IOperationOptions, IOperationFactory, OperationSelector } from '../operations/OperationSelector';
```

File: apps/rush-lib/src/logic/test/ProjectTaskSelector.test.ts [hunk 10]
```diff
@@ -3,39 +3,39 @@
...
-    name: task.name,
-    isCacheWriteAllowed: task.runner.isCacheWriteAllowed,
-    dependencies: Array.from(task.dependencies, (dep: Task) => dep.name)
+    name: operation.name,
+    isCacheWriteAllowed: operation.runner.isCacheWriteAllowed,
+    dependencies: Array.from(operation.dependencies, (dep: Operation) => dep.name)
   };
 }
 
 describe('ProjectTaskSelector', () => {
```
Comment (iclanton) [inline apps/rush-lib/src/logic/test/ProjectTaskSelector.test.ts:28]:
```suggestion
describe(OperationSelector.name, () => {
```

**Review by iclanton — APPROVED** [review]
Looks overall good. Someone else should take another look since this is such a big change.

**Review by D4N14L — APPROVED** [review]

**Pull request merged**